web/docs/deployment/deployment-methods/cli.md

```diff
@@ -140,9 +142,93 @@ Using the `www` and `non-www` domains at the same time will require you to updat
 
 :::
 
+## Railway {#railway}
+
+### Prerequisites
+
+:::caution Railpack required
+
+Wasp CLI requires that Railpack is set as the default deployment builder **for client routing to work correctly**.
+
+Go to your [Railway account settings](https://railway.com/account/feature-flags) and enable "Default to Railpack".
+
+:::
+
+To deploy to Railway using Wasp CLI, make sure to:
+
+1. Create a [Railway](https://railway.com/) account,
+
+1. Install the [`railway` CLI](https://docs.railway.com/guides/cli#installing-the-cli) on your machine.
+
+### Deploying
+
+Using the Wasp CLI, you can easily deploy a new app to Railway with a single command:
+
+```shell
+wasp deploy railway launch my-wasp-app
+```
+
+<small>
+  Please do not CTRL-C or exit your terminal while the commands are running.
+</small>
+
+Keep in mind that:
+
+1. Your project name (for example `my-wasp-app`) must be unique across all your Railway projects or deployment will fail. This is a current limitation of the Wasp CLI and Railway integration ([#2926](https://github.com/wasp-lang/wasp/issues/2926)).
+
+1. If you are a member of multiple Railway organizations, the CLI will prompt you to select the organization under which you want to deploy your app.
+
+Under the hood, `wasp deploy railway launch` is the same as running:
+
+```shell
+wasp deploy railway setup my-wasp-app
+wasp deploy railway deploy my-wasp-app
+```
+
+The project name is used as a base for your server and client service names on Railway:
+
+- `my-wasp-app-client`
+- `my-wasp-app-server`
+
+Railway doesn't allow setting the database service name using the Railway CLI, so it will always be named `Postgres`.
+
+When you run the `launch` command, Wasp CLI sets some [required environment variables](../../project/env-vars.md#wasp-server-env-vars) in the server app automatically.
+
+If you have any additional environment variables that your app needs, read how to set them in the [API Reference](#railway-environment-variables).
+
+### Using a Custom Domain For Your App {#railway-custom-domain}
+
+Setting up a custom domain is a three-step process:
+
+1. You need to add your domain to your Railway client service:
+
+    - Go into the [Railway dashboard](https://railway.app/dashboard).
+    - Select your project (for example `my-wasp-app`).
+    - Click on the client service (for example `my-wasp-app-client`).
+    - Go to the `Settings` tab and click `Custom Domain`.
+    - Enter your domain name (for example `mycoolapp.com`) and port `8080`.
+    - Click `Add Domain`.
+
+2. You need to add the DNS records for your domain:
+
+   _This will depend on your domain provider, but it should be a matter of adding a CNAME record with the values provided by Railway._
+
+3. You need to set your new client URL as the `WASP_WEB_CLIENT_URL` environment variable (for example `https://mycoolapp.com`) for your **server service** in the Railway dashboard.
```

Comment:
```suggestion
3. To avoid [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Guides/CORS) errors, you need to set your new client URL as the `WASP_WEB_CLIENT_URL` environment variable (for example `https://mycoolapp.com`) for your **server service** in the Railway dashboard.
```